sympy/core/function.py

```diff
@@ -3096,5 +3096,58 @@ def nfloat(expr, n=15, exponent=False):
         lambda x: x.func(*nfloat(x.args, n, exponent)),
         lambda x: isinstance(x, Function)))
 
+def log_concave(func, *args):
+    """Checks if the function passed in the argument is logarithamically concave.
+
+    Examples
+    ========
+
+    >>> from sympy.core.function import *
+    >>> from sympy.abc import x, y
+    >>> from sympy import exp, cos
+    >>> log_concave(cos(x), x)
+    False
+    >>> log_concave(exp(x), x)
+    True
+    """
+    from sympy import solve_univariate_inequality as s
+
+    syms = args[:]
+
+    if len(syms) > 1:
+        raise NotImplementedError("Log concavity check for multivariate functions is not implemented yet.")
+
+    var = syms[0]
+    ratio = func * func.diff(var).diff(var) / (func.diff(var)**2)
+    if s(ratio > 1, var):
+        return False
+    return True
+
+def log_convex(func, *args):
+    """Checks if the function passed in the argument is logarithamically convex.
+
+    Examples
+    ========
+
+    >>> from sympy.core.function import *
+    >>> from sympy.abc import x
+    >>> from sympy import exp, cos
+    >>> log_convex(cos(x), x)
+    False
+    >>> log_convex(exp(x), x)
+    True
+    """
+    from sympy import solve_univariate_inequality as s
+
+    syms = args[:]
+
+    if len(syms) > 1:
+        raise NotImplementedError("Log convexity check for multivariate functions is not implemented yet.")
+
+    var = syms[0]
+    ratio = func * func.diff(var).diff(var) / (func.diff(var)**2)
+    if s(ratio < 1, var):
```

Comment:
What is `s`?

Reply:
s is `solve_univariate_equality`.

Reply:
I think it may be better to just write the whole thing out. Will increase the readability. (This may have been partly the point of @asmeurer as well.)

Reply:
Sure I will do the same.